concepts.tex

```diff
@@ -472,30 +472,39 @@
 \begin{itemdecl}
 template <class T, class U>
 concept bool Assignable =
-  CommonReference<const T&, const U&> && requires(T&& t, U&& u) {
-    { std::forward<T>(t) = std::forward<U>(u) } -> Same<T&>;
+  is_lvalue_reference<T>::value && // \seebelow
+  CommonReference<
+    const remove_reference_t<T>&,
+    const remove_reference_t<U>&> &&
+  requires(T t, U&& u) {
+    { t = std::forward<U>(u) } -> Same<T>&&;
   };
 \end{itemdecl}
 
 \begin{itemdescr}
 \pnum
-Let \tcode{t} be an lvalue of type \tcode{T}, and \tcode{R} be the
-type \tcode{remove_reference_t<U>}. If \tcode{U} is an lvalue reference
-type, let \tcode{v} be an lvalue of type \tcode{R};
-otherwise, let \tcode{v} be an rvalue of type \tcode{R}.
-Let \tcode{uu} be a distinct object of type \tcode{R} such that
-\tcode{uu} is equal to \tcode{v}.
-\tcode{Assignable<T, U>} is satisfied only if
+Let \tcode{t} be an lvalue which refers to an object \tcode{o} such that
+\tcode{decltype((t))} is \tcode{T}, and \tcode{u} an expression such that
+\tcode{decltype((u))} is \tcode{U}. Let \tcode{u2} be a distinct object that is
+equal to \tcode{u}. \tcode{Assignable<T, U>} is satisfied only if
 
 \begin{itemize}
-\item \tcode{addressof(t = v) == addressof(t)}.
-\item After evaluating \tcode{t = v}, \tcode{t} is equal to \tcode{uu} and:
+\item \tcode{addressof(t = u) == addressof(o)}.
+
+\item After evaluating \tcode{t = u}, \tcode{t} is equal to \tcode{u2} and:
+
 \begin{itemize}
-\item If \tcode{v} is a non-\tcode{const} rvalue, its resulting
-state is valid but unspecified~(\cxxref{lib.types.movedfrom}).
-\item Otherwise, \tcode{v} is not modified.
+\item If \tcode{u} is a non-\tcode{const} xvalue, the resulting state of the
+object to which it refers is valid but unspecified~(\cxxref{lib.types.movedfrom}).
+
+\item Otherwise, if \tcode{u} is a glvalue, the object to which it refers is not
+modified.
 \end{itemize}
 \end{itemize}
+
+\pnum
+There need not be any subsumption relationship between \tcode{Assignable<T, U>}
+and \tcode{is_lvalue_reference<T>::value}.
```

Comment:
I plan to handle poor line wrapping in a separate pass after incorporating all of the PRs. I've added a note to #455 to be sure to check these two occurrences specifically.